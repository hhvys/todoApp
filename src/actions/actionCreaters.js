import * as actionTypes from './actionTypes'

let nextId = localStorage.getItem('nextId') || 2;

const storeNextId = () => localStorage.setItem('nextId', nextId.toString());

export function searchQuery(query) {
	return {
		type: actionTypes.SEARCH_QUERY,
		query
	}
}

export function toggleVisibilityFilter() {
	return {
		type: actionTypes.TOGGLE_VISIBILITY_FILTER
	};
}

export function changeActiveTab(tabId) {
	return {
		type: actionTypes.CHANGE_ACTIVE_TAB,
		tabId
	};
}

export function toggleSideBar() {
	return {
		type: actionTypes.TOGGLE_SIDEBAR
	};
}

export function toggleModal() {
	return {
		type: actionTypes.TOGGLE_MODAL
	}
}

export function toggleEdit() {
	return {
		type: actionTypes.TOGGLE_EDIT
	}
}

export function addTab(tabName) {
	nextId++;
	storeNextId();
	return {
		type: actionTypes.ADD_TAB,
		tabId: nextId,
		tabName,
	}
}

export function deleteTab(tabId) {
	return {
		type: actionTypes.DELETE_TAB,
		tabId
	}
}

export function changeTabName(tabId, tabName) {
	console.log(tabId, tabName);
	return {
		type: actionTypes.CHANGE_TAB_NAME,
		tabId,
		tabName
	}
}

export function addTodo(tabId, text) {
	nextId++;
	storeNextId();
	return {
		type: actionTypes.ADD_TODO,
		tabId,
		todoId: nextId,
		text
	}
}

export function addStarredTodo(tabId, text) {
	nextId++;
	storeNextId();
	return {
		type: actionTypes.ADD_STARRED_TODO,
		tabId,
		todoId: nextId,
		text
	}
}

export function toggleTodo(tabId, todoId) {
	return {
		type: actionTypes.TOGGLE_TODO,
		tabId,
		todoId
	}
}

export function activeTodo(tabId, todoId) {
	return {
		type: actionTypes.ACTIVE_TODO,
		tabId,
		todoId
	}
}

export function copyTab(fromId) {
	nextId++;
	storeNextId();
	return {
		type: actionTypes.COPY_TAB,
		fromId,
		toId: nextId
	}
}

export function toggleStarTodo(tabId, todoId) {
	return {
		type: actionTypes.STAR_TOGGLE_TODO,
		tabId,
		todoId
	}
}

export function changeSorting(sortBy) {
	return {
		type: actionTypes.CHANGE_SORT,
		sortBy
	}
}

export function toggleDropDown() {
	return {
		type: actionTypes.TOGGLE_DROPDOWN
	}
}