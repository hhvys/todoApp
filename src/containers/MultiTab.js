import {connect} from 'react-redux';
import MultiTabView from '../components/organisms/MultiTabView/MultiTabView';
import {
	addStarredTodo,
	changeActiveTab,
	changeSorting,
	searchQuery,
	toggleStarTodo,
	toggleTodo
} from "../actions/actionCreators";
import {getTabsWithInfo} from "../reducers/tabs/tabs";
import {getSearchQuery} from "../reducers/searchQuery";
import {getActiveTab} from "../reducers/activeTab";
import {getCollapsedSideBar} from "../reducers/collapsedSideBar";

export const getFilteredTabs = (state) => {
	let tabs = getTabsWithInfo(state);
	tabs = state.searchQuery.length ?
		tabs
			.map(tab => ({
				...tab,
				todos: tab
					.todos
					.filter(todo => !todo.completed && (new RegExp(state.searchQuery).test(todo.text)))
			})) :
		tabs
			.map(tab => ({
				...tab,
				todos: tab
					.starredTodos
					.filter(todo => !todo.completed)
			}));

	return tabs.filter(tab => tab.todos.length);
};

const mapStateToProps = (state) => {
	return {
		tabs: getFilteredTabs(state),
		searchQuery: getSearchQuery(state),
		activeTab: getActiveTab(state),
		collapsed: getCollapsedSideBar(state)
	};
};

const mapDispatchToProps = (dispatch) => {

	return {
		onFooterIconClick: (tabId, todoId) => {
			dispatch(toggleStarTodo(tabId, todoId));
			dispatch(changeSorting(undefined, tabId));
		},
		onHeaderIconClick: (tabId, todoId) => {
			dispatch(toggleTodo(tabId, todoId));
			dispatch(changeSorting(undefined, tabId));
		},
		onButtonClick: (tabId) => {
			dispatch(changeActiveTab(tabId));
			dispatch(searchQuery(''));
			dispatch(changeSorting());
		},
		onInputSubmit: (value) => {
			dispatch(addStarredTodo(value));
			// dispatch(changeSorting());
		}
	};
};

const MultiTab = connect(
	mapStateToProps,
	mapDispatchToProps
)(MultiTabView);

export default MultiTab;
