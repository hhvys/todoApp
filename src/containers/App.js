import {connect} from 'react-redux';
import AppView from '../components/App';
import {STARRED_ID} from "../actions/actionTypes";
import {getSearchQuery} from "../reducers/searchQuery";
import {getActiveTab} from "../reducers/activeTab";

const mapStateToProps = (state) => ({
	multiView: getSearchQuery(state).length !== 0 || getActiveTab(state) === STARRED_ID
});

const mapDispatchToProps = () => ({
});

const App = connect(
	mapStateToProps,
	mapDispatchToProps
)(AppView);

export default App;
