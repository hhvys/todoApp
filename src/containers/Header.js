import {connect} from 'react-redux';
import HeaderView from '../components/organisms/Header/Header';
import {changeSorting} from "../actions/actionCreaters";
import {getTabs} from "../reducers/tabs/tabs";
import {getSearchQuery} from "../reducers/searchQuery";
import {getActiveTab} from "../reducers/activeTab";
import {getCollapsedSideBar} from "../reducers/collapsedSideBar";


const mapStateToProps = (state) => {
	const tabs = getTabs(state);
	const searchQuery = getSearchQuery(state);
	const activeTab = getActiveTab(state);
	return ({
		headerContent: searchQuery ?
			searchQuery :
			tabs
				.find(tab => tab.tabId === activeTab)
				.tabName,
		collapsed: getCollapsedSideBar(state)
	})
};

const mapDispatchToProps = (dispatch) => {

	return ({
		onDropDownTabClick: (sortBy) => dispatch(changeSorting(sortBy)),
	})
};


const Header = connect(
	mapStateToProps,
	mapDispatchToProps
)(HeaderView);

export default Header;
