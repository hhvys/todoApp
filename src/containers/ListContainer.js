import {connect} from 'react-redux';
import ListContainerView from '../components/organisms/ListContainerView/ListContainerView';
import {changeActiveTab, changeSorting, searchQuery, toggleModal} from "../actions/actionCreaters";
import {getActiveTab} from "../reducers/activeTab";
import {getTabInfo} from "../reducers/tabs/tabInfo";

const getFooterContent = (inCompletedTodos) => {
	if(inCompletedTodos <= 0)
		return '';
	return inCompletedTodos > 99 ? '99+' : inCompletedTodos
};

const getTabsWithFooterContent = (state) => {
	let tabs = getTabInfo(state);
	return tabs.map(tab => {
		return {
			...tab,
			footerContent: getFooterContent(tab.inCompletedTodos)
		}
	});
};


const mapStateToProps = (state) => {

	return {
		tabs: getTabsWithFooterContent(state),
		activeTab: getActiveTab(state),
	}
};

const mapDispatchToProps = (dispatch) => {
	return {
		onClick: (tabId) => {
			dispatch(changeActiveTab(tabId));
			dispatch(searchQuery(''));
			dispatch(changeSorting());
		},
		onFooterSymbolClick: (tabId) => {
			dispatch(toggleModal(tabId));
		}
	}
};

const ListContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(ListContainerView);

export default ListContainer;
