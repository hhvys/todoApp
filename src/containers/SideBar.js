import {connect} from 'react-redux';
import SideBarView from '../components/organisms/SideBar/SideBarView';
import {searchQuery, toggleModal, toggleSideBar} from "../actions/actionCreaters";
import {getTabs} from "../reducers/tabs/tabs";
import {getActiveTab} from "../reducers/activeTab";
import {getCollapsedSideBar} from "../reducers/collapsedSideBar";

const mapStateToProps = (state) => {

	return {
		activeTab: getActiveTab(state),
		tabs: getTabs(state),
		collapsed: getCollapsedSideBar(state)
	};

};

const mapDispatchToProps = (dispatch) => {
	return {
		onSearchChange: (query) => dispatch(searchQuery(query)),
		onCreateClick: () => dispatch(toggleModal()),
		collapseSideBar: () => dispatch(toggleSideBar())
	};
};

const SideBar = connect(
	mapStateToProps,
	mapDispatchToProps
)(SideBarView);

export default SideBar;

