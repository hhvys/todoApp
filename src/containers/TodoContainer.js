import {connect} from 'react-redux';
import TodoView from '../components/organisms/TodoView/TodoView';
import {addTodo, toggleStarTodo, toggleTodo, toggleVisibilityFilter} from "../actions/actionCreaters";
import {getTabs} from "../reducers/tabs";
import {getActiveTab} from "../reducers/activeTab";
import {getSortBy} from "../reducers/sortBy";
import {getCollapsedSideBar} from "../reducers/collapsedSideBar";

const mapStateToProps = (state) => {
	const tabs = getTabs(state);
	const activeTab = getActiveTab(state);
	const todos = tabs
		.find(tab => tab.tabId === activeTab)
		.todos;
	return ({
		todos,
		activeTab,
		showCompleted: tabs
			.find(tab => tab.tabId === activeTab)
			.showCompletedTodo,
		sortBy: getSortBy(state),
		collapsed: getCollapsedSideBar(state)
	});
};

const mapDispatchToProps = (dispatch) => {
	return ({
		onHeaderSymbolClick: (tabId, todoId) => dispatch(toggleTodo(tabId, todoId)),
		onFooterSymbolClick: (tabId, todoId) => dispatch(toggleStarTodo(tabId, todoId)),
		onButtonClick: (tabId) => dispatch(toggleVisibilityFilter(tabId)),
		onInputSubmit: (tabId, value) => dispatch(addTodo(tabId, value))
	});
};

const TodoContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(TodoView);

export default TodoContainer;
