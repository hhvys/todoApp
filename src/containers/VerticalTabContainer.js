import {connect} from 'react-redux';
import TabContainerView from '../components/organisms/TabContainerView/TabContainerView';
import {changeActiveTab, searchQuery, toggleEdit, toggleModal} from "../actions/actionCreaters";

const mapStateToProps = (state) => {
	return {
		tabs: state.tabs,
		activeTab: state.activeTab,
	}
};

const mapDispatchToProps = (dispatch) => {
	return {
		onClick: (tabId) => {
			dispatch(changeActiveTab(tabId));
			dispatch(searchQuery(''));
		},
		onFooterSymbolClick: (tabId) => {
			dispatch(toggleModal());
			dispatch(toggleEdit());
		}
	}
};

const VerticalTabContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(TabContainerView);

export default VerticalTabContainer;