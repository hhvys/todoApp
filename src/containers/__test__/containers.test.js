import React from 'react';
import configureMockStore from 'redux-mock-store';
import {INITIAL_STATE, STATE_WITH_TABS} from "../../reducers/__test__/__fixtures__/reducers.fixtures";
import * as _ from 'lodash';
import App from "../App";
import {
	ADD_STARRED_TODO, ADD_TODO,
	CHANGE_ACTIVE_TAB, CHANGE_SORT,
	INBOX_ID,
	SEARCH_QUERY,
	STAR_TOGGLE_TODO,
	STARRED_ID,
	TOGGLE_MODAL, TOGGLE_SIDEBAR, TOGGLE_TODO, TOGGLE_VISIBILITY_FILTER
} from "../../actions/actionTypes";
import Header from "../Header";
import SideBar from "../SideBar";
import thunk from 'redux-thunk';
import ListContainer from '../ListContainer';
import MultiTab, {getFilteredTabs} from '../MultiTab';
import {SEARCHED_FILTERED_TABS, STARRED_FILTERED_TABS} from "./__fixtures__/containers.fixtures";
import {SORT_CREATION} from "../../components/atoms/Icons/constants";
import {v4} from 'node-uuid';
import {searchQuery} from "../../actions/actionCreaters";
import TodoContainer from "../TodoContainer";

jest.mock('node-uuid', () => ({v4: jest.fn(() => 1)}));

const mockStore = configureMockStore([thunk]);
const shallowWithStore = (component, store) => {
	const context = {
		store,
	};
	return shallow(component, { context });
};
describe('containers', () => {
	describe('App', () => {

		it('should render MultiTabView', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			initialState.searchQuery = 'searchQuery';
			const store = mockStore(initialState);
			const component = shallowWithStore(<App />, store);
			expect(component.dive().find("Connect(MultiTabView)").length).toBe(1);
			expect(toJSON(component.dive())).toMatchSnapshot();
		});

		it('should not render TodoContainer', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			initialState.searchQuery = 'searchQuery';
			const store = mockStore(initialState);
			const component = shallowWithStore(<App />, store);
			expect(component.dive().find("Connect(TodoContainer)").length).toBe(0);
		});

		it('should render MultiTabView', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			initialState.activeTab = STARRED_ID;
			const store = mockStore(initialState);
			const component = shallowWithStore(<App />, store);
			expect(component.dive().find("Connect(MultiTabView)").length).toBe(1);
		});

		it('should not render TodoContainer', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			initialState.activeTab = STARRED_ID;
			const store = mockStore(initialState);
			const component = shallowWithStore(<App />, store);
			expect(component.dive().find("Connect(TodoContainer)").length).toBe(0);
		});

		it('should not render MultiTabView', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			const store = mockStore(initialState);
			const component = shallowWithStore(<App />, store);
			expect(component.dive().find("Connect(MultiTabView)").length).toBe(0);
		});

		it('should render TodoContainer', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			const store = mockStore(initialState);
			const component = shallowWithStore(<App />, store);
			expect(component.dive().find("Connect(TodoContainer)").length).toBe(1);
			expect(toJSON(component.dive())).toMatchSnapshot();
		});

	});

	describe('Header', () => {
		it('should render searchQuery as header', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			initialState.searchQuery = 'searchQuery';
			const store = mockStore(initialState);
			const component = shallowWithStore(<Header />, store);
			const headerContent = component.dive().find(".headerContent");
			expect(headerContent.text()).toBe(initialState.searchQuery);
			expect(toJSON(component.dive())).toMatchSnapshot();
		});

		it('should render tabName as header', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			const store = mockStore(initialState);
			const component = shallowWithStore(<Header />, store);
			const headerContent = component.dive().find(".headerContent");
			expect(headerContent.text()).toBe('Inbox');
		});

		it('should have collapsed class', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			initialState.collapsedSideBar = true;
			const store = mockStore(initialState);
			const component = shallowWithStore(<Header />, store);
			const header = component.dive().find(".collapsed");
			expect(header.length).toBe(1);
			expect(toJSON(component.dive())).toMatchSnapshot();
		});

		it('should dispatch changeSorting', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			const store = mockStore(initialState);
			const component = shallowWithStore(<Header />, store);
			component.instance().selector.props.onDropDownTabClick();
			const expectedActions = [
				{ type: 'CHANGE_SORT', sortBy: 'SORT_CREATION' }
			];
			expect(store.getActions()).toEqual(expectedActions);
		});
	});

	describe('ListContainer', () => {

		it('should render initialState', () => {
			const store = mockStore(INITIAL_STATE);
			const component = shallowWithStore(<ListContainer />, store).dive();
			expect(toJSON(component)).toMatchSnapshot();
		});

		it('should dispatch toggleModal', () => {
			const store = mockStore(STATE_WITH_TABS);
			const component = shallowWithStore(<ListContainer />, store);
			component.instance().selector.props.onFooterSymbolClick();
			const expectedActions = [
				{type: TOGGLE_MODAL}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		it('should dispatch changeActiveTab, searchQuery', () => {
			const store = mockStore(STATE_WITH_TABS);
			const component = shallowWithStore(<ListContainer />, store);
			component.instance().selector.props.onClick(INBOX_ID);
			const expectedActions = [
				{
					type: CHANGE_ACTIVE_TAB,
					tabId: INBOX_ID,
				},
				{
					type: SEARCH_QUERY,
					query: ''
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

	});

	describe('MultiTab', () => {

		it('should render initialState', () => {
			const store = mockStore(STATE_WITH_TABS);
			store.activeTab = STARRED_ID;
			const component = shallowWithStore(<MultiTab />, store).dive();
			expect(toJSON(component)).toMatchSnapshot();
		});

		it('should dispatch toggleStarTodo', () => {
			const store = mockStore(STATE_WITH_TABS);
			const tabId = '9ad18f5c-d3b8-46ff-a54e-69bba97f1aa1';
			const todoId = 'a88ff79a-b113-407e-9298-bfb4811b44c5';
			const component = shallowWithStore(<MultiTab />, store);
			component.instance().selector.props.onFooterSymbolClick(tabId, todoId);
			const expectedActions = [
				{
					tabId,
					todoId,
					type: STAR_TOGGLE_TODO
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		it('should dispatch toggleTodo', () => {
			const store = mockStore(STATE_WITH_TABS);
			const tabId = '9ad18f5c-d3b8-46ff-a54e-69bba97f1aa1';
			const todoId = 'a88ff79a-b113-407e-9298-bfb4811b44c5';
			const component = shallowWithStore(<MultiTab />, store);
			component.instance().selector.props.onHeaderSymbolClick(tabId, todoId);
			const expectedActions = [
				{
					tabId,
					todoId,
					type: TOGGLE_TODO
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		it('should dispatch changeActiveTab, searchQuery', () => {
			const store = mockStore(STATE_WITH_TABS);
			const tabId = '9ad18f5c-d3b8-46ff-a54e-69bba97f1aa1';
			const component = shallowWithStore(<MultiTab />, store);
			component.instance().selector.props.onButtonClick(tabId);
			const expectedActions = [
				{
					tabId,
					type: CHANGE_ACTIVE_TAB
				},
				{
					query: '',
					type: SEARCH_QUERY
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		it('should dispatch addStarredTodo, changeSorting', () => {
			const store = mockStore(STATE_WITH_TABS);
			const text = 'text';
			const component = shallowWithStore(<MultiTab />, store);
			component.instance().selector.props.onInputSubmit(text);
			const expectedActions = [
				{
					type: ADD_STARRED_TODO,
					tabId: INBOX_ID,
					text,
					todoId: v4()
				},
				{
					type: CHANGE_SORT,
					sortBy: SORT_CREATION
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		describe('getFilteredTabs', () => {
			it('should return filtered tabs based on searchQuery', () => {
				const initialState = _.cloneDeep(STATE_WITH_TABS);
				initialState.searchQuery = 'a';
				const store = mockStore(initialState);
				expect(getFilteredTabs(store.getState())).toEqual(SEARCHED_FILTERED_TABS);
			});

			it('should return filtered starred tabs', () => {
				const initialState = _.cloneDeep(STATE_WITH_TABS);
				initialState.activeTab = STARRED_ID;
				const store = mockStore(initialState);
				expect(getFilteredTabs(store.getState())).toEqual(STARRED_FILTERED_TABS);
			});

		})

	});

	describe('SideBar', () => {
		it('should render', () => {
			const store = mockStore(INITIAL_STATE);
			const component = shallowWithStore(<SideBar />, store).dive();
			expect(toJSON(component)).toMatchSnapshot();
		});

		it('should render collapsedSideBar', () => {
			const initialState = _.cloneDeep(INITIAL_STATE);
			initialState.collapsedSideBar = true;
			const store = mockStore(initialState);
			const component = shallowWithStore(<SideBar/>, store).dive();
			expect(toJSON(component)).toMatchSnapshot();
			expect(component.find('CollapsedSideBar').length).toBe(1);
		});

		it('should not render collapsedSideBar', () => {
			const store = mockStore(INITIAL_STATE);
			const component = shallowWithStore(<SideBar />, store).dive();
			expect(component.find('CollapsedSideBar').length).toBe(0);
		});

		it('should dispatch searchQuery', () => {
			const searchQuery = 'searchQuery';
			const store = mockStore(STATE_WITH_TABS);
			const component = shallowWithStore(<SideBar />, store);
			component.instance().selector.props.onSearchChange(searchQuery);
			const expectedActions = [
				{
					type: SEARCH_QUERY,
					query: searchQuery
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		it('should dispatch toggleModal', () => {
			const store = mockStore(STATE_WITH_TABS);
			const component = shallowWithStore(<SideBar />, store);
			component.instance().selector.props.onCreateClick();
			const expectedActions = [
				{
					type: TOGGLE_MODAL,
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		it('should dispatch toggleSideBar', () => {
			const store = mockStore(STATE_WITH_TABS);
			const component = shallowWithStore(<SideBar />, store);
			component.instance().selector.props.collapseSideBar();
			const expectedActions = [
				{
					type: TOGGLE_SIDEBAR,
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

	});

	describe('TodoContainer', () => {

		it('should render', () => {
			const store = mockStore(INITIAL_STATE);
			const component = shallowWithStore(<TodoContainer />, store).dive();
			expect(toJSON(component)).toMatchSnapshot();
		});

		it('should dispatch toggleTodo', () => {
			const tabId = INBOX_ID;
			const todoId = 'a383d653-0fc9-4873-bf88-e4ee57885632';
			const store = mockStore(STATE_WITH_TABS);
			const component = shallowWithStore(<TodoContainer />, store);
			component.instance().selector.props.onHeaderSymbolClick(tabId, todoId);
			const expectedActions = [
				{
					type: TOGGLE_TODO,
					tabId,
					todoId
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		it('should dispatch toggleVisibilityFilter', () => {
			const tabId = INBOX_ID;
			const store = mockStore(STATE_WITH_TABS);
			const component = shallowWithStore(<TodoContainer />, store);
			component.instance().selector.props.onButtonClick(tabId);
			const expectedActions = [
				{
					type: TOGGLE_VISIBILITY_FILTER,
					tabId,
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		it('should dispatch toggleStarTodo, changeSorting', () => {
			const tabId = INBOX_ID;
			const todoId = 'a383d653-0fc9-4873-bf88-e4ee57885632';
			const store = mockStore(STATE_WITH_TABS);
			const component = shallowWithStore(<TodoContainer />, store);
			component.instance().selector.props.onFooterSymbolClick(tabId, todoId);
			const expectedActions = [
				{
					type: STAR_TOGGLE_TODO,
					tabId,
					todoId
				},
				{
					type: CHANGE_SORT,
					sortBy: SORT_CREATION
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

		it('should dispatch addTodo, changeSorting', () => {
			const tabId = INBOX_ID;
			const text = 'text';
			const store = mockStore(STATE_WITH_TABS);
			const component = shallowWithStore(<TodoContainer />, store);
			component.instance().selector.props.onInputSubmit(tabId, text);
			const expectedActions = [
				{
					type: ADD_TODO,
					tabId,
					text,
					todoId: v4()
				},
				{
					type: CHANGE_SORT,
					sortBy: SORT_CREATION
				}
			];
			expect(store.getActions()).toEqual(expectedActions);
		});

	});

});
