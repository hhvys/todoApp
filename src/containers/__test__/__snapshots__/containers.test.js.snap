// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers App should render MultiTabView 1`] = `
<div
  className="full-size"
>
  <Connect(SideBarView)
    className="full-height fixed"
  />
  <Connect(Header)
    style={
      Object {
        "backgroundColor": "#678865",
      }
    }
  />
  <Connect(MultiTabView) />
  <Connect(ModalView) />
</div>
`;

exports[`containers App should render TodoContainer 1`] = `
<div
  className="full-size"
>
  <Connect(SideBarView)
    className="full-height fixed"
  />
  <Connect(Header)
    style={
      Object {
        "backgroundColor": "#678865",
      }
    }
  />
  <Connect(TodoContainer) />
  <Connect(ModalView) />
</div>
`;

exports[`containers Header should have collapsed class 1`] = `
<div>
  <div
    className=" collapsed header fixed full-width d-flex align-items-center justify-content-between"
  >
    <div
      className="headerContent pl-4"
    >
      Inbox
    </div>
    <div
      className="dropdownIcons d-flex align-items-center justify-content-between"
    >
      <DropDownIcon
        iconSymbol="SORT"
        iconText="Sort"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`containers Header should render searchQuery as header 1`] = `
<div>
  <div
    className="  header fixed full-width d-flex align-items-center justify-content-between"
  >
    <div
      className="headerContent pl-4"
    >
      searchQuery
    </div>
    <div
      className="dropdownIcons d-flex align-items-center justify-content-between"
    >
      <DropDownIcon
        iconSymbol="SORT"
        iconText="Sort"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`containers ListContainer should render initialState 1`] = `
<div
  className=" flex-column justify-content-start align-items-center"
>
  <RowComponent
    active={true}
    className="tab"
    footerClass="footerLogo"
    footerContent={false}
    footerSymbol={false}
    headerClass="logo inboxLogo"
    headerSymbol="INBOX"
    key="0"
    mainContent="Inbox"
    onClick={[Function]}
    onFooterSymbolClick={[Function]}
  />
</div>
`;

exports[`containers MultiTab should render initialState 1`] = `
<div
  className="multiTabView  mr-3"
>
  <InputWithLabel
    className="input"
    label="+"
    onSubmit={[Function]}
    placeholder="Add a starred to-do in Inbox..."
  />
  <div
    key="0"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="INBOX"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STARRED"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="5ccdce50-5266-4872-aa7f-c8e5e7d4e74c"
      mainContent="5"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STARRED"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="6d0db783-1bcb-40ab-ab48-12a0ae8aa773"
      mainContent="2"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
  <div
    key="9ad18f5c-d3b8-46ff-a54e-69bba97f1aa1"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="1ST TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STARRED"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="a88ff79a-b113-407e-9298-bfb4811b44c5"
      mainContent="tab 1 6"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
  <div
    key="e26ee1de-944d-4b3a-88cc-1464fe53f32b"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="2ND TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STARRED"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="85b1e538-724c-4c72-ad8c-ae51bf4db5af"
      mainContent="tab 2 1"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
</div>
`;

exports[`containers SideBar should render 1`] = `
<div
  className="sideBar  d-flex flex-column align-items-center justify-content-center"
  tabs={
    Array [
      Object {
        "starredTodos": Array [],
        "tabId": 0,
        "tabName": "Inbox",
        "todos": Array [],
      },
      Object {
        "starredTodos": Array [],
        "tabId": 1,
        "tabName": "Starred",
        "todos": Array [],
      },
    ]
  }
>
  <SearchBar
    className="searchBar"
    collapseSideBar={[Function]}
    inputValue=""
    onInputChange={[Function]}
  />
  <Connect(ListContainerView)
    className="verticalTabContainer"
    footerSymbol="PENCIL"
    headerSymbol="LIST_SYMBOL"
  />
  <RowComponent
    className="footer"
    headerClass="footerLogo"
    headerSymbol="PLUS"
    mainContent="Create List"
    onClick={[Function]}
  />
</div>
`;

exports[`containers SideBar should render collapsedSideBar 1`] = `
<CollapsedSideBar
  collapseSideBar={[Function]}
/>
`;

exports[`containers TodoContainer should render 1`] = `
<TodoView
  activeTab={0}
  collapsed={false}
  onButtonClick={[Function]}
  onFooterSymbolClick={[Function]}
  onHeaderSymbolClick={[Function]}
  onInputSubmit={[Function]}
  todos={Array []}
/>
`;
